app/V1Module/security/Policies/UploadedFilePermissionPolicy.php

```diff
@@ -58,6 +58,30 @@ public function isAuthorOfSupplementaryFileExercises(Identity $identity, Uploade
     return FALSE;
   }
 
+
+  public function isInGroupUserSupervises(Identity $identity, UploadedFile $file)
+  {
+    if (!($file instanceof SupplementaryExerciseFile)) {
+      return false;
+    }
+
+    $user = $identity->getUserData();
+    if ($user === null) {
+      return false;
+    }
+
+    foreach ($file->getExercises() as $exercise) {
+      foreach ($exercise->getGroups() as $group) {
+        if ($group->isAdminOrSupervisorOfSubgroup($user)) {
+          return true;  // The user can assign one of the corresponding exercises in hir group.
```

Comment:
Is `hir` a typo or a gender-neutral pronoun?

Reply:
The latter.